src/ts/common/ui/modal.ts

```diff
@@ -93,31 +97,60 @@ const createTextContainer = (text: string, subTexts?: string[]) => {
 	return container;
 };
 
-const addOnClick = (element: HTMLElement, exit: () => void, onClick?: () => Promise<void>) => {
+const createOnClick = (exit: () => void, onClick?: () => Promise<void>) => {
 	const callback = onClick ?? (() => Promise.resolve());
-	element.addEventListener("click", () => callback().finally(exit));
+	return () => callback().finally(exit);
 };
 
-const dismissModals = (): void => activeOverlays.forEach((modal) => modal.dispatchEvent(new Event("click")));
```

Comment:
If it's not `exitable`, it won't have a click event, so you should just call the `onCancel` which was decorated to end with a dismissal